src/Microsoft.ML.Tokenizers/Tokenizer.cs

```diff
@@ -230,6 +295,56 @@ private enum ModelEncoding
                                                                 { "gpt2", ModelEncoding.GPT2 }
                                                             };
```

Comment:
Not directly related to this PR, but it looks like CreateByNameXx only works for Tiktoken. Is there anything that would make sense to add here for the other models M.ML.Tokenizers supports? I'm asking less because I care about those models and more because I want to make sure Tokenizer.CreateByNameXx is scalable to things beyond Tiktoken for future proofing.

Reply:
My though is Tokenizer.CreateByNameXx may work with other models but will not be generic enough to cover all possible scenarios. As you see the new created factory method in this PR is exposed directly in Tiktoken because the parameter like the vocab stream is not necessary be enough for all other models. We can decide to move CreateByNameXx to Tiktoken and new Tokenizer models can have its own factory methods according to whatever parameters it work with. If you agree I can apply this change in this PR as it will affect the integration to copilot.

Reply:
If it's specific to Tiktoken, I agree we should move it to Tiktoken. As it stands, it looks like it should generally apply to any tokenizer, but it doesn't. If in the future there are other models that support this, or we get to the point of supporting something like autotokenizer, we could add something back to the main type.